oauth_client.go

```diff
@@ -134,6 +135,9 @@ type OAuthClientCreateOptions struct {
 
 	// Required: The VCS provider being connected with.
 	ServiceProvider *ServiceProviderType `jsonapi:"attr,service-provider"`
+
+	// Optional: AgentPoolID to associate the VCS Provider with, for PrivateVCS support
+	AgentPoolID *string `jsonapi:"attr,agent-pool-id,omitempty"`
```

Comment:
This should also be a relation. What does the API expect? agent_pool_id attribute? Is it too late to change it to be more conventional?

Reply:
Yes, the API expects agent pools external_id as an attribute. Thats how the API is called from the frontend.

Reply:
That does not seem to be documented already, so is this new API? Can it be moved to the relationships key? This would bring it into convention with other APIs, like [creating an agent pool with allowed workspaces](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/agents#request-body) and [creating a run that references a configuration version and workspace](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/run#create-a-run) and [updating a workspace to be in a different project](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/workspaces#update-a-workspace)

Reply:
Yes, thats a new API for Private VCS project, which has only the UI changes in scope. We built just one API to support the integration tests. 
Thanks for sharing the other APIs, I'll look into those and follow the convention.